### PR TITLE
[release-4.12] OCPBUGS-23020: Introduce upgrading label to block concurrent upgrades

### PR DESCRIPTION
**Pull request by jrvaldes**
This is a manual cherry-pick of https://github.com/openshift/windows-machine-config-operator/pull/1901

### CODE REVIEW

**Comment by mtnbikenc** [comment]
/hold
Until after 6.0.3

File: test/e2e/upgrade_test.go [hunk 1]
```diff
@@ -217,6 +219,24 @@ func TestUpgrade(t *testing.T) {
 
 	// test that any workloads deployed on the node have not been broken by the upgrade
 	t.Run("Workloads ready", tc.testWorkloadsAvailable)
+	t.Run("Parallel Upgrades Checker", tc.testParallelUpgradesChecker)
+
```
Comment (wgahnagl) [inline test/e2e/upgrade_test.go:223]: super nit: there's an extra newline here that also existed in the original PR

**Review by wgahnagl — COMMENTED** [review]
/lgtm

**Comment by jrvaldes** [comment]
/hold 

depends on to fix azure-e2e-operator test :
- https://github.com/openshift/windows-machine-config-operator/pull/2002

**Comment by jrvaldes** [comment]
/hold cancel 

blocking PR merged

**Comment by jrvaldes** [comment]
/hold 

need [WINC-1191](https://issues.redhat.com/browse/WINC-1191) for validation

**Comment by jrvaldes** [comment]
blocked by 

- https://github.com/openshift/release/pull/47918

**Comment by jrvaldes** [comment]
/hold cancel

blocker pr merged.

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Pull request merged**